backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -84,7 +84,37 @@ export class SegmentService {
       .andWhere({ id })
       .getOne();
 
-    return segmentDoc;
+    if (!segmentDoc) {
+      return undefined;
+    }
+
+    const segmentsUsedList = [];
```

Comment:
This logic is checking direct sub-segment which is not an entirely correct solution. 
@ppratikcr7 can you help Jay with this

Reply:
new query had been created and this code has been removed

Reply:
The private segments are also stored in the Segment table. The question is why are you filtering private segments in getAllSegmentsAndSubsegments?

Reply:
Private segments are called ahead in another function where Experiment's segments are called and we check it in nested loop, in that we get details from the subsegments.